WalletWasabi/Services/UpdateChecker.cs

```diff
@@ -39,7 +41,11 @@ protected override async Task ActionAsync(CancellationToken cancel)
 		if (newUpdateStatus != UpdateStatus)
 		{
 			UpdateStatus = newUpdateStatus;
-			UpdateStatusChanged?.Invoke(this, newUpdateStatus);
+			UpdateManager.UpdateStatusChangedAsync(newUpdateStatus);
```

Comment:
I see the point of simplifying the subscription inside `UpdateManager`, but is unconventional and couples the classes together needlessly. UpdateChecker should not know about `UpdateManager`. `UpdateManager` should use the `UpdateStatusChanged` event.

Reply:
This way, we dodge to add anothe event + subscribe/unsubscribe, and we can also easily call `UpdateManager.DeletePossibleLefotver()` to remove unecessary files if we have the new version installed.

Reply:
Events are our friends. Think about the software from a higher level. To avoid chaos, you need to structure your software somehow. Step back one and try to write small specifications for every class/method you are writing. Try to keep that spec, so you will know what the responsibilities of that specific code part are. Whenever you break that, you probably add work for the future (technical debt), that somebody will have to pay and refactor your class to get rid of the unnecessary coupling of separate classes and responsibilities. Even if there is only one subscriber it is worth to have 2 classes because they are doing different things.